auth/auth.ts

```diff
@@ -105,19 +106,25 @@ export class MockAuth implements firebase.auth.Auth {
 
   private signIn(
     user: User,
-    additionalUserInfo: firebase.auth.AdditionalUserInfo | null = null
+    { isNewUser }: { isNewUser: boolean }
```

Comment:
Are we shifting away from `AdditionalUserInfo` or is temporary?

If the intention is to only have `isNewUser`, would it not be better to have it as a single value rather then an object?
Alternatively if the goal is to use `AdditionalUserInfo` later why not use a `Partial<>` for the time being?

Reply:
> Are we shifting away from `AdditionalUserInfo` or is temporary?

Neither. If you look at `UserCredential`, the constructor infers most of it.


> would it not be better to have it as a single value rather then an object?

My intention was probably to avoid a [boolean trap](https://ariya.io/2011/08/hall-of-api-shame-boolean-trap).
However, I fell in this very trap in the `UserCredential` constructor... 😅

Reply:
🤔   That was an interesting read. Could we turn `{ isNewUser: boolean }` into some type of interface?